runtime/src/rpc.rs

```diff
@@ -217,15 +219,17 @@ impl InterBtcParachain {
         Self::new(ws_client, signer, shutdown_tx).await
     }
 
-    async fn get_fresh_nonce(&self) -> u32 {
+    async fn get_fresh_nonce(&self) -> Result<u32, Error> {
```

Comment:
Not sure if we want this to return an error since it may bypass the existing error handling, worth checking

Reply:
IF we don't pass out an error back to the caller, we need to add a unwrap/error log.

Reply:
What are the possible errors now incurred? I think in `notify_retry` we will match some specific errors if we will check specifically if we can retry or propagate the error.

Reply:
It will pass out a subxt runtime error defined [here](https://github.com/interlay/interbtc-clients/blob/6b5933f01a4ef0b602a56a9acd31965b80dc89e7/runtime/src/error.rs#L81). I have moved the [get_fresh_nonce](https://github.com/interlay/interbtc-clients/blob/6b5933f01a4ef0b602a56a9acd31965b80dc89e7/runtime/src/rpc.rs#L268C34-L268C49) method inside  [match timeout](https://github.com/interlay/interbtc-clients/blob/6b5933f01a4ef0b602a56a9acd31965b80dc89e7/runtime/src/rpc.rs#L269) in latest commit.

And i don't think there is a need to update the retry/propagate checks as the [create_signed_with_nonce](https://github.com/interlay/interbtc-clients/blob/6b5933f01a4ef0b602a56a9acd31965b80dc89e7/runtime/src/rpc.rs#L273) method returns the same subxt error type, and has been already handle in the master branch so same checks should work for `get_fresh_nonce` method as well.